spinalcordtoolbox/process_seg.py

```diff
@@ -154,69 +188,174 @@ def compute_shape(segmentation, angle_correction=True, centerline_path=None, par
+                                                     tform.inverse,
+                                                     output_shape=current_patch_im.shape,
+                                                     order=1,
+                                                     )
+        else:
+            current_patch_im_scaled = current_patch_im
+
+        # compute PCA and get center or mass based on segmentation; centermass_src: [RL, AP] (assuming RPI orientation)
+        coord_src, pca_src, centermass_src = compute_pca(current_patch_scaled)
+        # Finds the angle of the image
+        # TODO: explore different sigma values for the HOG method, i.e., the influence how far away pixels will vote for the orientation.
+        # TODO: double-check if sigma is in voxel or mm units.
+        # TODO: do we want to use the same sigma for all slices? As the spinal cord sizes vary across the z-axis.
+        angle_hog, conf_src = find_angle_hog(current_patch_im_scaled, centermass_src,
+                                             px, py, angle_range=40)    # 40 is taken from registration.algorithms.register2d_centermassrot
+
+        z_indices.append(iz)
+        angle_hog_values.append(angle_hog)
+        centermass_values.append(centermass_src)
+
+    # Apply regularization to HOG angles along the z-axis if filter_size > 0
+    if filter_size > 0 and len(z_indices) > 0:
+        angle_hog_regularized = gaussian_filter1d(np.array(angle_hog_values), filter_size)
+        if verbose == 2:
+            create_regularized_hog_angle_plot(
+                np.array(z_indices), filter_size,
+                np.array(angle_hog_values), angle_hog_regularized
+            )
+    # If filter_size <= 0, then just use the original angles
+    else:
+        angle_hog_regularized = angle_hog_values
+
+    # Now compute shape properties using the new angles
+    for iz, angle_hog, centermass_src in zip(z_indices, angle_hog_regularized, centermass_values):
+        current_patch_scaled = current_patches[iz]['patch']
+        angle_AP_rad = current_patches[iz]['angle_AP_rad']
+        angle_RL_rad = current_patches[iz]['angle_RL_rad']
+
+        # Compute shape properties with regularized angle_hog
+        shape_property = _properties2d(current_patch_scaled, [px, py], iz, angle_hog=angle_hog, verbose=verbose)
+
+        if shape_property is not None:
+            # Add custom fields
+            shape_property['centermass_x'] = centermass_src[0]
+            shape_property['centermass_y'] = centermass_src[1]
+            shape_property['angle_hog'] = -angle_hog * 180.0 / math.pi     # degrees, and change sign to match negative if left rotation
+            shape_property['angle_AP'] = angle_AP_rad * 180.0 / math.pi     # convert to degrees
+            shape_property['angle_RL'] = angle_RL_rad * 180.0 / math.pi     # convert to degrees
+            shape_property['length'] = pz / (np.cos(angle_AP_rad) * np.cos(angle_RL_rad))
```

Comment:
I missed this before, but these properties don't actually change when `angle_hog` is provided, right? 

If they are identical, then we can probably remove these lines from the `_properties_image` function (since they will have already been computed in `compute_shape`).

Reply:
done here: https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4958/commits/a8daf0d1dcbfeab57c7ab5ea5b4700b32be29f63